### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by kmewhort**
**Reviewer Approvals**:
- [x] @dimerman
- [x] @justintanner

**Notes**:

The meat of this is in  `lib/griddler/ses/adapter.rb` (and apologies about the .idea file clutter that ended up in `master`, deleted here).  Unfortunately no specs at this point....pretty hard to write ones that aren't trivial outside of Griddler.  This is consistent with Griddler's other plugins though.  Manual integration testing checks out.

### CODE REVIEW

File: lib/griddler/ses/adapter.rb [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      end
+
+      def confirm_sns_subscription_request
+        HTTParty.get sns_json['SubscribeURL']
+      end
+
+      def self.raw_request
+        # TODO: this is an ugly hack using introspection to get the request from the calling controller; should update Griddler
+        # to provide the full request context
+        raw_request = binding.of_caller(2).eval('request')
```
Comment (kmewhort) [inline lib/griddler/ses/adapter.rb:137]: I'll be discussing a PR to Griddler once this PR passes muster.

File: lib/griddler/ses/adapter.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        tempfile.rewind
+        tempfile
+      end
+
+      def ensure_valid_notification_type!
+        raise "Invalid SNS notification type (\"#{notification_type}\", expecting Received" unless notification_type == 'Received'
+      end
+
+      def confirm_sns_subscription_request
+        HTTParty.get sns_json['SubscribeURL']
```
Comment (dimerman) [inline lib/griddler/ses/adapter.rb]: for gems, it's best practice to use Ruby's `Net::HTTP` - to keep it slim

File: lib/griddler/ses/adapter.rb [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      end
+
+      def attachment_files
+        # also based on griddler-sparkpost (https://github.com/PrestoDoctor/griddler-sparkpost, MIT license);
+        # AWS doesn't presently support sending the attachments from the message through SNS, but ready if they do!
+        message.attachments.map do |attachment|
+          ActionDispatch::Http::UploadedFile.new({
+            type: attachment.mime_type,
+            filename: attachment.filename,
+            tempfile: tempfile_for_attachment(attachment)
```
Comment (dimerman) [inline lib/griddler/ses/adapter.rb:111]: who closes the temp file ?

Comment (kmewhort) [inline lib/griddler/ses/adapter.rb:111]: They're left open for clients of Griddler to use in the hooks.  Ruby closes and deletes non-explicitly closed tempfiles on garbage collection.

**Comment by dimerman** [comment]
✔️  (can't tick the PR description)

**Comment by justintanner** [comment]
Code looks good to me.

**Pull request merged**